tsdb/shard.go

```diff
@@ -472,6 +506,11 @@ func (s *Shard) SeriesKeys(opt influxql.IteratorOptions) (influxql.SeriesList, e
 		return []influxql.Series{{Aux: auxFields}}, nil
 	}
 
+	s.mu.RLock()
```

Comment:
You could probably do something like the following to reduce some of the locking/nil code that is duplicated.
```
func (s *Shard) IsClosed() bool {
  s.mu.RLock()
  defer s.mu.RUnlock()
  return s.engine == nil
}
```

and then use:

```
if s.IsClosed() {
  return nil, ErrEngineClosed
}
```